indexer_app/utils.py

```diff
@@ -805,14 +806,67 @@ async def handle_add_nadabot_admin(data, receiverId):
 async def handle_add_factory_deployers(data, receiverId):
     logger.info(f"adding factory deployer...: {data}, {receiverId}")
     try:
-        factory = await PotFactory.objects.aget(id=receiverId)
-
+        factory = await PotFactory.objects.aget(account=receiverId)
         for acct in data["whitelisted_deployers"]:
             user, _ = await Account.objects.aget_or_create(id=acct)
             await factory.whitelisted_deployers.aadd(user)
     except Exception as e:
         logger.error(f"Failed to add factory whitelisted deployers, Error: {e}")
 
+async def handle_update_factory_whitelist_required(data, receiverId):
+    logger.info(f"updating factory whitelist required...: {data}, {receiverId}")
+    try:
+        factory = await PotFactory.objects.aget(account=receiverId)
+        factory.require_whitelist = data["require_whitelist"]
+        await factory.asave()
+    except Exception as e:
+        logger.error(f"Failed to update factory whitelist required, Error: {e}")
+    
+async def handle_set_factory_configs(data, receiverId):
+    logger.info(f"setting factory protocol recipient...: {data}, {receiverId}")
+    try:
+        factory = await PotFactory.objects.aget(account=receiverId)
+        if data.get("protocol_fee_recipient_account"):
+            protocol_recipient, _ = await Account.objects.aget_or_create(id=data["protocol_fee_recipient_account"])
+            factory.protocol_fee_recipient = protocol_recipient
+        if data.get("protocol_fee_basis_points"):
+            factory.protocol_fee_basis_points = data["protocol_fee_basis_points"]
+        if data.get("require_whitelist"):
+            factory.require_whitelist = data["require_whitelist"]
+        if data.get("new_owner"):
+            new_owner, _ = await Account.objects.aget_or_create(id=data["new_owner"])
+            factory.owner = new_owner
+        if data.get("whitelisted_deployers"):
+            for acct in data["whitelisted_deployers"]:
+                user, _ = await Account.objects.aget_or_create(id=acct)
+                await factory.whitelisted_deployers.aadd(user)
+        await factory.asave()
+    except Exception as e:
+        logger.error(f"Failed to set factory protocol recipient, Error: {e}")
```

Comment:
i'm confused, this is all duplicate code from `PotFactory.update_configs`. if it's because you need async methods, have you tried wrapping the sync method in sync_to_async?